### DISCUSSION TITLE
How to return structured ApiError when extending as a framework

### DISCUSSION THREAD

**Discussion by thangngoc89** [pocketbase/pocketbase#4949] [Q&A] [answered]
I'm trying to return structured error data (at least a machine readable code) in a hook. 

Here is what I've tried so far:

```golang
import (
   validation "github.com/go-ozzo/ozzo-validation/v4"
)

return apis.NewBadRequestError(
  "human readable message",
  validation.NewError(
    "machine_code",
    "machine message"
  ),
)
```

I could see the nicely formatted "Human Readable Message.". However the `data` field is empty from the API response.

**Comment by ganigeorgiev** [comment, marked as answer]
The data value must be a `map[string]validation.Error` (or `validation.Errors`) in order to be publicly visible, aka. in other words you can try with:

```go
return apis.NewBadRequestError("human readable message", validation.Errors{
    // field -> error
    "machine": validation.NewError("machine_code", "machine message"),
})
```

**Reply by thangngoc89** [reply]
Thanks. That did it
